include/aklite.h

```diff
@@ -0,0 +1,178 @@
+  };
+  DownloadResult(Status status, std::string description = "") : status(status), description(std::move(description)) {}
+  Status status;
+  std::string description;
+};
+
+/**
+ * The response from an AkliteClient call to Install
+ */
+class InstallResult {
+ public:
+  enum class Status {
+    Ok,
+    NeedsCompletion,
+    Failed,
+  };
+  InstallResult(Status status, std::string description = "") : status(status), description(std::move(description)) {}
+  Status status;
+  std::string description;
+};
+
+/**
+ * AkliteClient provides an easy-to-use API for users wanting to customize
+ * the behavior of aktualiz-lite.
+ */
+class AkliteClient {
+ public:
+  /**
+   * Construct a client instance pulling in config files from the given
+   * locations. ex:
+   *
+   *   AkliteClient c(AkliteClient::CONFIG_DIRS)
+   *
+   * @param config_dirs The list of files/directories to parse sota toml from.
+   */
+  AkliteClient(const std::vector<boost::filesystem::path> &config_dirs);
+  /**
+   * Used for unit-testing purposes.
+   */
+  AkliteClient(std::shared_ptr<LiteClient> client) : client_(client) {}
+
+  /**
+   * Performs a "check-in" with the device-gateay. A check-in consists of:
+   *  1) ask device-gateway for a new root.json - normally a 404.
+   *  2) ask device-gateway for timestamp and snapshot metadata.
+   *  3) pull down a new targets.json if needed
+   */
+  CheckInResult CheckIn(std::string hwid = "") const;
+
+  /**
+   * Return the active aktualizr-lite configuration.
+   */
+  boost::property_tree::ptree GetConfig() const;
+
+  /**
+   * Report network and hwinfo to server. This only reports data if the
+   * information has changed locally.
+   */
+  void SendTelemetry() const;
+
+  /**
+   * Report the full aktualizr-lite active configuration to the
+   * device-gateway. Should be called once during start-up .
+   */
+  void ReportAkliteConfiguration() const;
+
+  /**
+   * Return the Target currently running on the system.
+   */
+  TufTarget GetCurrent() const;
+
+  /**
+   * Download and verify the content of the given target.
+   */
+  DownloadResult Download(const TufTarget &t, const std::string &correlation_id, std::string reason = "");
+
+  /**
+   * Install the Target
+   */
+  InstallResult Install(const TufTarget &t, const std::string &correlation_id);
+
+  /**
+   * Check if the Target has been installed but failed to boot. This would
+   * make this be considered a "rollback target" and one we shouldn't consider
+   * installing.
+   */
+  bool IsRollback(const TufTarget &t) const;
+
+  /**
+   * Default files/paths to search for sota toml when configuration client.
+   */
+  std::vector<boost::filesystem::path> CONFIG_DIRS = {"/usr/lib/sota/conf.d", "/var/sota/sota.toml",
+                                                      "/etc/sota/conf.d/"};
+
+  /**
+   * Return a random UUID to be used for a Download/Install operation
+   */
+  static std::string GenerateCorrelationId(const TufTarget &t);
```

Comment:
It could be an internal stuff, aklite client could generate it internally if a client doesn't specify it in the Download() and  Install() calls?

Reply:
the same value needs to be in both calls to correlate the download and install.

Reply:
what makes a client set the same value of a cor ID for these both methods?